### PR TITLE
fix from_xml: return python dictionary

### LINKED ISSUE #114: from_xml filter returns a string instead of a python dictionary
##### SUMMARY
The `from_xml` filter returns [a string](https://github.com/ansible-collections/ansible.utils/blob/f8d0e4f39909769f376e513c61ef0613a497eafd/plugins/plugin_utils/from_xml.py#L46) instead of returning a python dictionary as [its documentation reports](https://github.com/ansible-collections/ansible.utils/blob/f8d0e4f39909769f376e513c61ef0613a497eafd/plugins/filter/from_xml.py#L19) and as the [`from_json`](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L1458) and `from_yaml` filters do.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
from_xml

##### ANSIBLE VERSION
```paste below
ansible [core 2.11.3] 
  config file = /home/user/workspace/ansible-playbooks-2/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/user/workspace/ansible-playbooks-2/.direnv/python-3.8.10/lib/python3.8/site-packages/ansible
  ansible collection location = /home/user/workspace/ansible-playbooks-2/collections
  executable location = /home/user/workspace/ansible-playbooks-2/.direnv/python-3.8.10/bin/ansible
  python version = 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
  jinja version = 3.0.1
  libyaml = True
```

##### COLLECTION VERSION
```paste below
# /home/user/workspace/ansible-playbooks-2/collections/ansible_collections
Collection    Version
------------- -------
ansible.utils 2.4.2  

```

##### CONFIGURATION

```paste below
AGNOSTIC_BECOME_PROMPT(/home/user/workspace/ansible-playbooks-2/ansible.cfg) = False
ANSIBLE_FORCE_COLOR(env: ANSIBLE_FORCE_COLOR) = True
CALLBACKS_ENABLED(/home/user/workspace/ansible-playbooks-2/ansible.cfg) = ['timer', 'ansible.posix.profile_tasks']
COLLECTIONS_PATHS(/home/user/workspace/ansible-playbooks-2/ansible.cfg) = ['/home/user/workspace/ansible-playbooks-2/collections']
DEFAULT_CONNECTION_PLUGIN_PATH(/home/user/workspace/ansible-playbooks-2/ansible.cfg) = ['/home/user/workspace/ansible-playbooks-2/connection_plugins']
DEFAULT_FORKS(/home/user/workspace/ansible-playbooks-2/ansible.cfg) = 10
DEFAULT_HOST_LIST(env: ANSIBLE_INVENTORY) = ['/home/user/workspace/ansible-playbooks-2/inventories/test']
DEFAULT_TIMEOUT(/home/user/workspace/ansible-playbooks-2/ansible.cfg) = 10
DEFAULT_VAULT_PASSWORD_FILE(/home/user/workspace/ansible-playbooks-2/ansible.cfg) = /home/user/workspace/ansible-playbooks-2/inventories/vault_password.sh
DISPLAY_SKIPPED_HOSTS(/home/user/workspace/ansible-playbooks-2/ansible.cfg) = False
HOST_KEY_CHECKING(env: ANSIBLE_HOST_KEY_CHECKING) = False
INVENTORY_ANY_UNPARSED_IS_FAILED(/home/user/workspace/ansible-playbooks-2/ansible.cfg) = True
INVENTORY_ENABLED(/home/user/workspace/ansible-playbooks-2/ansible.cfg) = ['host_list', 'ini', 'constructed']
INVENTORY_IGNORE_EXTS(/home/user/workspace/ansible-playbooks-2/ansible.cfg) = ['~', '.orig', '.bak', '.cfg', '.retry', '.pyc', '.pyo', 'LICENSE', '.md', '.txt', 'secrets.yml', 'vars.yml', 'ssh_private_key']
PLAYBOOK_DIR(env: ANSIBLE_PLAYBOOK_DIR) = /home/user/workspace/ansible-playbooks-2
RETRY_FILES_ENABLED(/home/user/workspace/ansible-playbooks-2/ansible.cfg) = True
RETRY_FILES_SAVE_PATH(/home/user/workspace/ansible-playbooks-2/ansible.cfg) = /tmp
TASK_TIMEOUT(env: ANSIBLE_TASK_TIMEOUT) = 180
VARIABLE_PRECEDENCE(/home/user/workspace/ansible-playbooks-2/ansible.cfg) = ['all_inventory', 'groups_inventory', 'all_plugins_play', 'groups_plugins_play', 'all_plugins_inventory', 'groups_plugins_inventory']

```

##### OS / ENVIRONMENT

Ubuntu 20.04.3 LTS (Focal Fossa)

##### STEPS TO REPRODUCE

```yaml
- hosts: localhost
  diff: true
  gather_facts: false

  tasks:
    - debug:
        msg: "{{ (some_xml | ansible.utils.from_xml)['root']['leaf'] }}"
      vars:
        some_xml: |
          <?xml version="1.0" encoding="UTF-8"?>
          <root>
              <leaf>
                <attribute>foobar</attribute>
              </leaf>
          </root>
```

##### EXPECTED RESULTS
This should print 
```
"msg": {                 
    "attribute": "foobar"
}                        
```

##### ACTUAL RESULTS

```paste below
fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'str object' has no attribute 'root'\n\nThe error appears to be in '/home/john/workspace_tagpay/ansible-playbooks-2/foo.tmp.yml': line 6, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n  tasks:\n    - debug:\n      ^ here\n"}
```

### PR DESCRIPTION
**Pull request by eberna**
##### SUMMARY

Fixes #114 

##### ISSUE TYPE

Bugfix Pull Request

##### COMPONENT NAME

ansible.utils.from_xml
##### ADDITIONAL INFORMATION

the filter ansible.utils.from_xml returns a json, instead of a python dictionary (as stated in the documentation). 
The json dump part has been removed and unit test has been modified to reflect this change.

### CODE REVIEW

**Comment by Ruchip16** [comment]
@eberna thank you for this contribution, could you please add changelog?

**Comment by eberna** [comment]
> @eberna thank you for this contribution, could you please add changelog?

Thanks to you! Changelog added.  :smile:

File: changelogs/fragments/fix_from_xml.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+breaking_changes:
```
Comment (Ruchip16) [inline changelogs/fragments/fix_from_xml.yaml:2]: hey i dont think these are breaking changes, could you please update this to minor changes?

**Review by Ruchip16 — CHANGES_REQUESTED** [review]

**Comment by Ruchip16** [comment]
> > @eberna thank you for this contribution, could you please add changelog?
> 
> Thanks to you! Changelog added. 😄

added a minor comment, thanks!

File: changelogs/fragments/fix_from_xml.yaml (hunk 1 above)
Comment (eberna) [reply changelogs/fragments/fix_from_xml.yaml]: sure, done.

**Review by Ruchip16 — APPROVED** [review]

**Comment by Ruchip16** [comment]
thanks for your contribution @eberna :clap:

**Pull request merged**